build.gradle.kts

```diff
@@ -0,0 +1,329 @@
+
+plugins {
+    java
+    `maven-publish`
+    `java-gradle-plugin`
+    id("net.researchgate.release") version "2.6.0"
+    id("com.github.sherter.google-java-format") version "0.8"
+    checkstyle
+    jacoco
+}
+
+repositories {
+    mavenCentral()
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
+    withSourcesJar()
+    withJavadocJar()
+}
+
+group = "com.google.cloud.tools"
+
+dependencies {
+    implementation(localGroovy())
+    implementation(gradleApi())
+    implementation("com.google.cloud.tools:appengine-plugins-core:0.9.8")
+
+    testImplementation("commons-io:commons-io:2.4")
+    testImplementation("junit:junit:4.12")
+    testImplementation("org.hamcrest:hamcrest-library:1.3")
+    testImplementation("org.mockito:mockito-core:2.23.4")
+}
+
+tasks {
+    wrapper {
+        gradleVersion = "6.8.2"
+    }
+    // See https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html#sub:disabling-gmm-publication
+    // For now, disable module metadata files since they need extra testing.
+    // Improper metadata might cause issues when consuming the dependency in Gradle
+    withType<GenerateModuleMetadata>().configureEach {
+        enabled = false
+    }
+    jar {
+        manifest {
+            attributes(
+                "Implementation-Title" to project.name,
+                "Implementation-Version" to project.version,
+                "Built-By" to providers.systemProperty("user.name"),
+                "Built-Date" to `java.time`.Instant.now().toString(),
+                "Built-JDK" to providers.systemProperty("java.version"),
+                "Built-Gradle" to gradle.gradleVersion
+            )
+        }
+    }
+}
+
+/* GRADLE PLUGINS */
+gradlePlugin {
+    plugins {
+        create("appengine") {
+            id = "$group.appengine"
+            implementationClass = "com.google.cloud.tools.gradle.appengine.AppEnginePlugin"
+            displayName = "App Engine Gradle Plugin"
+            description =
+                "This Gradle plugin provides tasks to build and deploy Google App Engine applications."
+        }
+        create("appengine-appenginewebxml") {
+            id = "$group.appengine-appenginewebxml"
+            implementationClass =
+                "com.google.cloud.tools.gradle.appengine.standard.AppEngineStandardPlugin"
+            displayName = "App Engine Gradle Plugin"
+            description =
+                "This Gradle plugin provides tasks to build and deploy Google App Engine applications."
+        }
+        create("appengine-appyaml") {
+            id = "$group.appengine-appyaml"
+            implementationClass =
+                "com.google.cloud.tools.gradle.appengine.appyaml.AppEngineAppYamlPlugin"
+            displayName = "App Engine Gradle Plugin"
+            description =
+                "This Gradle plugin provides tasks to build and deploy Google App Engine applications."
```

Comment:
I think these would be the only two we actually want to publish to maven central. All other naming is deprecated?

Reply:
I kept all the plugins, just like they were declared in properties files.
If you want to drop some, that is doable, however, I've no idea which ones are used in the wild.

Reply:
Hah yeah they're all in the wild. Many years ago, it was proposed to publish a subset of these to the gradle-plugin portal. **Maybe it's not worth dealing with here**. But the last major user guide update has some info here: https://github.com/GoogleCloudPlatform/app-gradle-plugin/blob/master/USER_GUIDE.md#user-guide-2

Perhaps this can be dealt with after the refactor.

Reply:
Just in case, after this refactor, the plugin will work via `plugins { id(...) }` even if you publish to Central only (I know Google did not like to publish to Portal for some reason).

Reply:
@elfel for future reference, a lot of docs probably still point to "appengine" so maybe we keep 3... appengine, appengine-appyaml and appengine-appenginewebxml.